File: codingjs/maxBlock.js
function maxBlock(str: string): number {
  if (str.length === 0) return 0;
  if (str.length === 1) return 1;
  return maxBlockHelper(str.substr(1), str.substr(0, 1), 1, 0);
}
function maxBlockHelper(str: string, prevChar: string, count: number, maxCount: number): number {
  if (str === "") {
    return count > maxCount ? count : maxCount;
  }
  if (str.substr(0, 1) === prevChar) {
    count++;
  } else {
    maxCount = count > maxCount ? count : maxCount;
    count = 1;
  }
  prevChar = str.substr(0, 1)
  return maxBlockHelper(str.substr(1), prevChar, count, maxCount);
}

File: codingbat/java/maxBlock.java
public int maxBlock(String str) {
  int count = 0;
  int maxCount = 0;
  if (str.length() <= 1) return str.length();
  
  char prevChar = str.charAt(0);
  for (int i = 0; i < str.length(); i++) {
    if (prevChar == str.charAt(i)) {
      count++;
      maxCount = count > maxCount ? count : maxCount;
    } else {
      maxCount = count > maxCount ? count : maxCount;
      count = 1;
    }
    prevChar = str.charAt(i);
  }
  return maxCount;
}

